src/components/molecules/Visualizer/Layers/store.ts

```diff
@@ -68,6 +73,72 @@ export class LayerStore {
     return l;
   }
 
+  #usedIds: string[] = [];
+  #newId = () => {
+    const genResId = () =>
+      "_xxxxxxxxxxxxxxxxxxxxxxxxx".replace(/[x]/g, function () {
+        return ((Math.random() * 16) | 0).toString(16);
+      });
+    let id;
+    do {
+      id = genResId();
+    } while (this.#usedIds.includes(id));
+    this.#usedIds.push(id);
+    return id;
+  };
+
```

Comment:
```suggestion
  #usedIds: Set<string> = new Set();
  #newId = () => {
    const genResId = () =>
      "_xxxxxxxxxxxxxxxxxxxxxxxxx".replace(/[x]/g, function () {
        return ((Math.random() * 16) | 0).toString(16);
      });
    let id;
    do {
      id = genResId();
    } while (this.#usedIds.has(id));
    this.#usedIds.add(id);
    return id;
  };

```

In this use case, Set is extremely faster than Array. Element retrieval in an array is O(n) while element retrieval in a Set is O(1). As the number of elements increases, the performance difference is obvious.